packages/core/android/build.gradle

```diff
@@ -205,6 +205,9 @@ dependencies {
   testImplementation "com.github.xgouchet.Elmyr:jvm:1.3.1"
   testImplementation "org.mockito.kotlin:mockito-kotlin:5.1.0"
   testImplementation "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
+  testImplementation "io.mockk:mockk:1.13.2" // For JVM unit tests
```

Comment:
why it is needed, if we already have Mockito added for mocking?

Reply:
It allows us to mock static classes with `mockkStatic`, effectively modifying the behaviour in the implementation without having to inject a mocked class. I can try removing it since I have introduced it for some test cases that I have now removed with the latest commits, but I think it can be really useful for implementing more tests in the future.

Reply:
usually mocking static value is some anti-pattern (because it will involve some dark bytecode manipulation) and should be used only if there is no possibility to organize the code in such a way, that mocking static is not needed.

Reply:
I have double-checked, the reason why we need is this:

```kotlin
mockkStatic(Looper::class)

val looper = mockk<Looper> {
    every { thread } returns Thread.currentThread()
}

every { Looper.getMainLooper() } returns looper
```

Because `ChoreographerCompat` internally uses `Looper.getMainLooper()` and it breaks the tests.

Reply:
Unfortunately `ChoreographerCompat` constructor is private, so the only way to access it is through its `getInstance()`. 

Before creating the instance, it executes `UiThreadUtil.assertOnUiThread()`, which is eventually calling:

```java
public static boolean isOnUiThread() {
    return Looper.getMainLooper().getThread() == Thread.currentThread();
}
```

Where `Looper.getMainLooper()` can be null and will be null in testing, causing:

> Cannot invoke "android.os.Looper.getThread()" because the return value of "android.os.Looper.getMainLooper()" is null

I don't see a clear way to avoid this issue from occurring, do you have any ideas?

Reply:
> does it make the job?

Looks good, let me check!

Reply:
It makes the job indeed 🚀 Thank you!